src/thor/triplegbuilder.cc

```diff
@@ -1543,13 +1544,21 @@ void TripLegBuilder::Build(
         edge_begin_info.trim = true;
         edge_begin_info.distance_along = start_pct;
         edge_begin_info.vertex = start_vrt;
+      } // No trimming needed
+      else if (!edge_begin_info.trim) {
+        edge_begin_info.distance_along = 0;
+        edge_begin_info.vertex = edge_shape.front();
```

Comment:
when we are not trimming either the front or the end we need to make sure we still set these things because they are used regardless below in the shape trimming function